src/main/java/oof/command/CalendarCommand.java

```diff
@@ -0,0 +1,155 @@
+        YearMonth yearMonth = YearMonth.of(year, month);
+        for (Task task : tasks.getTasks()) {
+            if (task instanceof Todo) {
+                Todo todo = (Todo) task;
+                String[] dateSplit = todo.getOn().split("-");
+                if (verifyTask(dateSplit)) {
+                    String description = todo.getLine();
+                    String time = "";
+                    int day = Integer.parseInt(dateSplit[INDEX_DAY]);
+                    addEntry(time, description, day);
+                }
+            } else if (task instanceof Deadline) {
+                Deadline deadline = (Deadline) task;
+                String[] dateTimeSplit = deadline.getBy().split(" ");
+                String[] dateSplit = dateTimeSplit[INDEX_DATE].split("-");
+                if (verifyTask(dateSplit)) {
+                    String description = deadline.getLine();
+                    String time = dateTimeSplit[INDEX_TIME];
+                    int day = Integer.parseInt(dateSplit[INDEX_DAY]);
+                    addEntry(time, description, day);
+                }
+            } else if (task instanceof Event) {
+                Event event = (Event) task;
+                String[] dateTimeSplit = event.getStartTiming().split(" ");
+                String[] dateSplit = dateTimeSplit[INDEX_DATE].split("-");
+                if (verifyTask(dateSplit)) {
+                    String description = event.getLine();
+                    String time = dateTimeSplit[INDEX_TIME];
+                    int day = Integer.parseInt(dateSplit[INDEX_DAY]);
+                    addEntry(time, description, day);
+                }
+            }
+        }
+        for (ArrayList<String[]> day : calendarTasks) {
+            day.sort(new SortByTime());
+        }
+        ui.printCalendar(yearMonth, calendarTasks);
+    }
+
+    /**
+     * Checks if task occurs in the month and year queried by user.
+     *
+     * @param dateSplit Array containing day, month and year.
+     * @return true if task month and year is equal to month and year queried by user.
+     */
+    public boolean verifyTask(String[] dateSplit) {
+        int month = Integer.parseInt(dateSplit[INDEX_MONTH]);
+        int year = Integer.parseInt(dateSplit[INDEX_YEAR]);
+        return this.month == month && this.year == year;
+    }
+
+    /**
+     * Adds an entry to the calendarTask ArrayList.
+     *
+     * @param time        Time of task.
+     * @param description Description of task.
+     * @param day         Day of task.
+     */
+    public void addEntry(String time, String description, int day) {
+        String[] entry = {time, description};
+        this.calendarTasks.get(day).add(entry);
+    }
+
+    /**
+     * Comparator to sort tasks by their time in ascending order.
+     */
+    class SortByTime implements Comparator<String[]> {
+        @Override
+        public int compare(String[] a, String[] b) {
```

Comment:
Missing Javadocs comment. All public and non-trivial private methods should have one.